insalan/pizza/views.py

```diff
@@ -0,0 +1,206 @@
+            return Response({"detail": _("Not found.")}, status=404)
+        timeslot = TimeSlot.objects.get(id=self.kwargs["pk"])
+        serializer = serializers.PizzaByTimeSlotSerializer(timeslot)
+        # only return the pizzas ordered
+        return Response(serializer.data)
+
+class TimeSlotList(generics.ListCreateAPIView):
+    """List all timeslots"""
+    pagination_class = PizzaPagination
+    queryset = TimeSlot.objects.all()
+    permission_classes = [permissions.IsAdminUser | ReadOnly]
+
+    def get_serializer_class(self):
+        if self.request.method == "GET":
+            return serializers.TimeSlotIdSerializer
+        return serializers.TimeSlotSerializer
+
+
+class TimeSlotListFull(generics.ListAPIView):
+    """List all timeslots"""
+    pagination_class = PizzaPagination
+    queryset = TimeSlot.objects.all()
+    serializer_class = serializers.TimeSlotSerializer
+    permission_classes = [ReadOnly]
+
+
+class TimeSlotDetail(generics.RetrieveAPIView, generics.DestroyAPIView):
+    """Find a timeslot by its id"""
+    queryset = TimeSlot.objects.all()
+    permission_classes = [permissions.IsAdminUser]
+
+    def get(self, request, *args, **kwargs):
+        if not TimeSlot.objects.filter(id=self.kwargs["pk"]).exists():
+            return Response({"detail": _("Not found.")}, status=404)
+        timeslot = TimeSlot.objects.get(id=self.kwargs["pk"])
+        serializer = serializers.TimeSlotSerializer(timeslot, context={"request": request}).data
+
+        order_serialized = []
+        for order in serializer["orders"]:
+            order = Order.objects.get(id=order)
+            order_serializer = serializers.OrderSerializer(order).data
+            order_serializer.pop("time_slot")
+            order_serialized.append(order_serializer)
+        serializer["orders"].clear()
+        serializer["orders"] = order_serialized
+
+        serializer.pop("player_product")
+        serializer.pop("staff_product")
+        serializer.pop("external_product")
+
+        return Response(serializer)
+
+class NextTimeSlot(generics.ListAPIView):
+    """Find the next timeslot"""
+    queryset = TimeSlot.objects.all()
+    permission_classes = [ReadOnly]
+
+    def get(self, request, *args, **kwargs):
+        # select incoming timeslot and currents timeslot
+        timeslot = TimeSlot.objects.filter(delivery_time__gte=timezone.now()).order_by("start")
+
+        serializers_data = []
+        for ts in timeslot:
+            serializer = serializers.TimeSlotSerializer(ts, context={"request": request}).data
+
+            serializer.pop("player_product")
+            serializer.pop("staff_product")
+            serializer.pop("external_product")
+
+            serializers_data.append(serializer)
+
+        return Response(serializers_data)
+
+class OrderList(generics.ListCreateAPIView):
+    """List all orders"""
+    queryset = Order.objects.all()
+    permission_classes = [permissions.IsAdminUser | ReadOnly]
```

Comment:
should we really let anyone read all current orders?

Reply:
well, the serializer used is only giving the ID of the orders but you are right, even this info could be too much.